### DISCUSSION TITLE
How to create a link that wraps an icon tag

### DISCUSSION THREAD

**Discussion by cfcoder** [wheels-dev/wheels#1396] [Q&A]
We use awesome font which uses tags like 
`<i class="fas fa-edit"></i>`
I'm not seeing anything in the docs about how to wrap a linkTo around one of these. 

Same question for an image tag <img>

**Comment by chapmandu** [comment]
Put your 'i' tag in the 'text' argument. You'll need to tinker with the quotes to get the syntax correct.

**Comment by cfcoder** [comment]
I tried doing that but wasn’t able to get it to accept it or do it.

**Reply by adamcameron** [reply]
It would be helpful - when posting a code question - if you posted what you've already tried, and how the outcome didn't match you expectation. Saves a lot of second-guessing.

**Reply by MvdO79** [reply]
I remember trying that also without any luck. At this moment for all links with icons I am using:

`<a href="#urlFor(controller="blogposts", action="myBlogPostEdit", key="#blogposts.pk_blogposts#")#" class="rate-review"><i class="im im-icon-Edit"></i> Edit</a>`

So if someone can post a working example of that "tinkering with the quotes" that would be great!

**Reply by neokoenig** [reply]
```
#linkTo(route="editSetting", encode="attributes", key=id, text="<i class='fa fa-edit'></i> " & "Edit", class="btn btn-sm btn-primary")#
```

**Reply by MvdO79** [reply]
Thanks @neokoenig , works great will add this to the docs.
Added an example to linkTo.
Also see some examples here: https://guides.cfwheels.org/2.5.0/displaying-views-to-users/linking-pages#extreme-example

@cfcoder is this example clarifying your additional question about using the imageTag in the text attribute:

https://guides.cfwheels.org/2.5.0/displaying-views-to-users/linking-pages#images-and-other-embedded-html-in-link-texts

**Comment by cfcoder** [comment]
Good point, Adam, I should disclose more. I’ll keep that in mind for future posts.
